src/ngraph/op/concat.cpp

```diff
@@ -93,6 +94,52 @@ shared_ptr<Node> op::Concat::copy_with_new_args(const NodeVector& new_args) cons
     return make_shared<Concat>(new_args, m_concatenation_axis);
 }
 
+std::vector<std::shared_ptr<op::Constant>> op::Concat::as_constants() const
+{
+    if (get_concatenation_axis() != 0)
+    {
+        return {};
+    }
+
+    size_t total_elements = 0;
+
+    for (size_t i = 0; i < get_input_size(); i++)
+    {
+        //
+        // For the time being we will only support int64 here, since that's all that's needed for
+        // static shape propagation.
+        //
+        if (get_input_element_type(i) != element::i64)
+        {
+            return {};
+        }
+        if (!(get_argument(i)->is_constant()))
+        {
+            return {};
+        }
+        if (get_input_shape(i).size() != 1)
+        {
+            return {};
+        }
+        total_elements += shape_size(get_input_shape(i));
+    }
+
+    std::vector<int64_t> values(total_elements);
+
+    size_t pos = 0;
+
+    for (size_t i = 0; i < get_input_size(); i++)
+    {
+        auto const_node = static_pointer_cast<op::Constant>(get_argument(i));
+        memcpy(values.data() + pos,
```

Comment:
Klocwork will probably flag this `memcpy` so please add a comment justifying why it's safe.

Reply:
Good call---I actually added an `assert` to be extra sure that the safety argument holds. :)